src/synapse/workload.py

```diff
@@ -0,0 +1,467 @@
+    """Run the Synapse command migrate_config.
+
+    Args:
+        container: Container of the charm.
+        charm_state: Instance of CharmState.
+
+    Raises:
+        CommandMigrateConfigError: something went wrong running migrate_config.
+    """
+    _check_server_name(container=container, charm_state=charm_state)
+    # TODO validate if is possible to use SDK instead of command  # pylint: disable=fixme
+    migrate_config_command = [SYNAPSE_COMMAND_PATH, COMMAND_MIGRATE_CONFIG]
+    migrate_config_result = _exec(
+        container,
+        migrate_config_command,
+        environment=get_environment(charm_state),
+    )
+    if migrate_config_result.exit_code:
+        logger.error(
+            "migrate config failed, stdout: %s, stderr: %s",
+            migrate_config_result.stdout,
+            migrate_config_result.stderr,
+        )
+        raise CommandMigrateConfigError(
+            "Migrate config failed, please review your charm configuration"
+        )
+
+
+def enable_metrics(container: ops.Container) -> None:
+    """Change the Synapse configuration to enable metrics.
+
+    Args:
+        container: Container of the charm.
+
+    Raises:
+        EnableMetricsError: something went wrong enabling metrics.
+    """
+    try:
+        config = container.pull(SYNAPSE_CONFIG_PATH).read()
+        current_yaml = yaml.safe_load(config)
+        metric_listener = {
+            "port": int(PROMETHEUS_TARGET_PORT),
+            "type": "metrics",
+            "bind_addresses": ["::"],
+        }
+        current_yaml["listeners"].extend([metric_listener])
+        current_yaml["enable_metrics"] = True
+        container.push(SYNAPSE_CONFIG_PATH, yaml.safe_dump(current_yaml))
+    except ops.pebble.PathError as exc:
+        raise EnableMetricsError(str(exc)) from exc
+
+
+def _get_mjolnir_config(access_token: str, room_id: str) -> typing.Dict:
+    """Create config as expected by mjolnir.
+
+    Args:
+        access_token: access token to be used by the mjolnir bot.
+        room_id: management room id monitored by the Mjolnir.
+
+    Returns:
+        Mjolnir configuration
+    """
+    with open("templates/mjolnir_production.yaml", encoding="utf-8") as mjolnir_config_file:
+        config = yaml.safe_load(mjolnir_config_file)
+        config["homeserverUrl"] = SYNAPSE_URL
+        config["rawHomeserverUrl"] = SYNAPSE_URL
+        config["accessToken"] = access_token
+        config["managementRoom"] = room_id
+        return config
+
+
+def create_mjolnir_config(container: ops.Container, access_token: str, room_id: str) -> None:
+    """Create mjolnir configuration.
+
+    Args:
+        container: Container of the charm.
+        access_token: access token to be used by the Mjolnir.
+        room_id: management room id monitored by the Mjolnir.
+
+    Raises:
+        CreateMjolnirConfigError: something went wrong creating mjolnir config.
+    """
+    try:
+        config = _get_mjolnir_config(access_token, room_id)
```

Comment:
Perhaps add the additional configuration in this function? Or just in-line `_get_mjolnir_config` since it seems to be pretty simple?

Reply:
Related: #52